CAIPs/caip-dagjws_cacao.md

```diff
@@ -51,23 +51,55 @@ Example 1: Protected JWS header with CACAO
 
 ### DagJWS with CACAO Construction
 
-Given JWS with CACAO described in prior section, follow the DAG-JOSE specification and implementations for the steps to construct a given JWS with CACAO header and payload into a DagJWS. 
+Given JWS with CACAO described in prior section, follow the DAG-JOSE specification and implementations for the steps to construct a given JWS with CACAO header and payload into a DagJWS. DagJWS is very similar to any JWS, except that the payload is a base64url encoded IPLD CID that references the JSON object payload. 
 
 ### DagJWS with CACAO Verification
 
 The following algorithm describes the steps required to determine if a given DagJWS with CACAO is valid. 
 
 1) Follow DAG-JOSE specification to transform a given DagJWS into a JWS. 
 
-2) Follow JWS specifications to determine if the given JWS is valid. Verifying that the given signature paired with ‘alg’ and ‘kid’ in the protected header is valid over the given payload. If invalid, an error MUST be raised. 
+2) Follow JWS specifications to determine if the given JWS is valid. Verifying that the given signature paired with `alg` and `kid` in the protected header is valid over the given payload. If invalid, an error MUST be raised. 
 
-3) Resolve the given URI in ‘cap’ parameter of the projected JWS header to a CACAO JSON object. Follow the CAIP-74 CACAO specification to determine if the given CACAO is valid. If invalid, an error MUST be raised. 
+3) Resolve the given URI in `cap` parameter of the projected JWS header to a CACAO JSON object. Follow the [CAIP-74 CACAO](https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-74.md) specification to determine if the given CACAO is valid. If invalid, an error MUST be raised. 
 
-4) Ensure that the ‘aud’ parameter of the CACAO payload is the same target as the ‘kid’ parameter in the JWS protected header. If they do not match, an error MUST be raised.
+4) Ensure that the `aud` parameter of the CACAO payload is the same target as the `kid` parameter in the JWS protected header. If they do not match, an error MUST be raised.
+
+### Example DagJWS with CACAO
+
+Example IPLD dag-jose encoded block, strings abbreviated. 
+
+```tsx
+{ 
+  cid: "bagcqcera2mews3mbbzs...quxj4bes7fujkms4kxhvqem2a",
+  value: {
+    jws: { 
+      link: CID("bafyreidkjgg6bi4juwx...lb2usana7jvnmtyjb4xbgwl6e"),
+      payload: "AXESIGpJjeCjiaWv...LKw6pIDQfTVrJ4SHlwmsvx", 
+      signatures: [
+        {
+          protected: "eyJhbGciOiJFZERTQSIsImNh...GU2djZEpLTmhYSDl4Rm9rdEFKaXlIQiJ9"
+          signature: "6usTYvu5KN0LFTQsWE9U-tqx...h60EgfvjL_rlAW7_tnQUl84sQyogpkLAQ"
+        }
+      ]
+    }
+  }
+}
+```
```

Comment:
Can use the full example with non-abbreviated strings?
So that you can test an implementation against it.

Reply:
would it make sense to 
1.) publish a sample/reference IPLD block and pin it for posterity, then
2.) link to it from the spec as an https://{cidv1}.ipfs.dweb.link/ link?
and/or
3.) put the entire IPLD block into an /assets/ folder in this repo and link to that from the spec as well to help people make sure they're fetching/deserializing/etc properly ?

i might be taking this whole `test vector` thing too far but i'm sure the folks at PL outercore would appreciate my asking for this to be extremely n00b-friendly to help with ipfs outreach :D

Reply:
test vectors are always nice, but not sure if makes sense for this spec here

This spec mostly describes how to use a cacao with a JWS, ie adding cacao in header and how to verify, it then also happens to be encoded in dag-jose, i think most of serialization/pathing/etc is entirely covered by dag-jose and would be covered by any tests vectors there and some verification is already covered by cacao spec, thanks for commenting!